bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostAsterism.scala

```diff
@@ -0,0 +1,368 @@
+
+    val enabled: OiwfsState  = Enabled
+    val disabled: OiwfsState = Disabled
+
+    val All = NonEmptyList(enabled, disabled)
+
+    implicit val EqualOiwfsState = Equal.equalA[OiwfsState]
+  }
+
+
+  /** GHOST targets are associated with a guiding state (enabled or disabled),
+    * referring to whether the dedicated guide fibers surrounding the science
+    * target should be used.
+    *
+    * There is a default guiding state based on magnitude, but this can be
+    * explicitly overridden.
+    */
+  final case class GhostTarget(target: SPTarget, explicitOiwfsState: Option[OiwfsState]) {
+
+    def coordinates(when: Option[Instant]): Option[Coordinates] =
+      target.getCoordinates(when.map(_.toEpochMilli))
+  }
+
+  object GhostTarget {
+
+    /** The magnitude at which OIWFS state is disabled by default at standard
+      * resolution.
+      */
+    val StandardResCutoff: Magnitude =
+      Magnitude(18.0, MagnitudeBand.B, None, MagnitudeSystem.Vega)
+
+    /** The magnitude at which OIWFS state is disabled by default at high
+      * resolution.
+      */
+    val HighResCutoff: Magnitude =
+      Magnitude(17.0, MagnitudeBand.B, None, MagnitudeSystem.Vega)
+
+    private def defaultOiwfsState(t: GhostTarget, cutoff: Magnitude): OiwfsState =
+      t.target.getMagnitude(MagnitudeBand.B).forall(_.value < cutoff.value) ? OiwfsState.enabled | OiwfsState.disabled
+
+    private def oiwfsState(t: GhostTarget, cutoff: Magnitude): OiwfsState =
+      t.explicitOiwfsState | defaultOiwfsState(t, cutoff)
+
+    /** Computes the OiwfsState for the given target in standard resolution mode. */
+    def standardResOiwfsState(t: GhostTarget): OiwfsState =
+      oiwfsState(t, StandardResCutoff)
+
+    /** Computes the OiwfsState for the given target in high resolution mode. */
+    def highResOiwfsState(t: GhostTarget): OiwfsState =
+      oiwfsState(t, HighResCutoff)
+  }
+
+
+  /** X-binning options. */
+  sealed abstract class XBinning(val intValue: Int) extends Product with Serializable {
+    override def toString: String =
+      intValue.toString
+  }
+
+  object XBinning {
+    case object One extends XBinning(1)
+    case object Two extends XBinning(2)
+
+    val one: XBinning = One
+    val two: XBinning = Two
+
+    val All = NonEmptyList(one, two)
+
+    implicit val EqualXBinning = Equal.equalA[XBinning]
+  }
+
+
+  /** Y-binning options. */
+  sealed abstract class YBinning(val intValue: Int) extends Product with Serializable {
+    override def toString: String =
+      intValue.toString
+  }
+
+  object YBinning {
+    case object One   extends YBinning(1)
+    case object Two   extends YBinning(2)
+    case object Four  extends YBinning(4)
+    case object Eight extends YBinning(8)
+
+    val one: YBinning   = One
+    val two: YBinning   = Two
+    val four: YBinning  = Four
+    val eight: YBinning = Eight
+
+    val All = NonEmptyList(one, two, four, eight)
+
+    implicit val EqualYBinning = Equal.equalA[YBinning]
```

Comment:
Would it make sense to define `Order` here and elsewhere when there's a natural ordering? I don't have a strong opinion but the UI might want to sort them rather than relying on the arbitrary order of `All`.